.github/workflows/ci.yml

```diff
@@ -0,0 +1,459 @@
+            sudo sh -c "echo 'memory_limit = $PHP_MEMORY_LIMIT' >> /etc/php/${{ matrix.php }}/apache2/php.ini"
+          fi
+
+          # Disable xdebug which greatly slow down unit testing
+          # Note: omitting xdebug from shivammathur/setup-php still results in xdebug being installed and enabled
+          if [ "${{ matrix.phpcoverage }}" == "true" ]; then
+            sudo sh -c "echo ';zend_extension=xdebug.so' > /etc/php/${{ matrix.php }}/mods-available/xdebug.ini"
+          fi
+          echo "PHP has been configured"
+
+      - name: Apt install additional requirements 
+        run: |
+          # apt install extra requirements as required
+          if [ "${{ matrix.endtoend }}" == "true" ]; then
+            sudo apt install -y software-properties-common
+            sudo add-apt-repository -y ppa:ondrej/php
+            sudo add-apt-repository -y ppa:ondrej/apache2
+            sudo apt update
+            sudo apt install -y libapache2-mod-php${{ matrix.php }}
+            # ubuntu-latest comes with a current version of google-chrome-stable and chromedriver
+          fi
+          if [[ "${{ github.repository }}" =~ /(spellcheck|recipe-authoring-tools)$ ]] || [ "${{ matrix.phpunit_suite }}" == "recipe-authoring-tools" ]; then
+            sudo apt install -y hunspell libhunspell-dev hunspell-en-us
+          fi
+
+      - name: Configure apache - endtoend test
+        if: ${{ matrix.endtoend == 'true' }}
+        run: |
+          # apache2 is installed and running by default in ubuntu
+          # update dir.conf to use index.php as the primary index doc
+          cat << EOF > __dir.conf
+          <IfModule mod_dir.c>
+              DirectoryIndex index.php index.html index.cgi index.pl index.xhtml index.htm
+          </IfModule>
+          EOF
+          sudo cp __dir.conf /etc/apache2/mods-enabled/dir.conf
+          rm __dir.conf
+          # this script will create a 000-default.conf file with the pwd as the DocumentRoot
+          cat << EOF > __000-default.conf
+          <VirtualHost *:80>
+            ServerAdmin webmaster@localhost
+            DocumentRoot PWD
+            <Directory PWD>
+              AllowOverride All
+              Require all granted
+            </Directory>
+            ErrorLog ${APACHE_LOG_DIR}/error.log
+            CustomLog ${APACHE_LOG_DIR}/access.log combined
+          </VirtualHost>
+          EOF
+          cat << EOF > __apache2.php
+          <?php
+          \$pwd = rtrim(trim(shell_exec('pwd')), '/');
+          \$s = file_get_contents('__000-default.conf');
+          \$s = str_replace('PWD', \$pwd, \$s);
+          file_put_contents('__000-default.conf', \$s);
+          EOF
+          php __apache2.php
+          sudo cp __000-default.conf /etc/apache2/sites-enabled/000-default.conf
+          rm __000-default.conf
+          sudo a2enmod rewrite
+          # run apache as 'runner:docker' instead of 'www-data:www-data'
```

Comment:
No link, I think I came up with this myself after checking what the current user and group was

This is to sort out all the file permission issues.  I don't think there's any risk of missing Silverstripe regressions by doing this.

Reply:
I wonder if we should set permissions according to our [server requirements](https://docs.silverstripe.org/en/4/getting_started/server_requirements/#filesystem-permissions) instead? That way we can both validate that the current docs are correct, as well as catch any sneaky attempts to need wider permissions than we say are required.

Reply:
I'd expect this to take a fair amount of time sort out so I don't want to do this right now, split this off as a separate card if you want

Reply:
I would expect it would be pretty straight forward - unless the thinking is that our required permissions list is extremely inaccurate? Or is there some other piece to the puzzle here I'm missing?

Reply:
Changing webserver permission and ensuring that behat still works inside a docker container is likely to take some time

Reply:
Fair enough. Created a new issue for this: https://github.com/silverstripe/gha-ci/issues/3